FogCreek/Glitch-Community
# Understood plot 1: Replace UserList & TeamList with ProfileList

## Links
* Remix link: https://understood-plot-1.glitch.me
* Manuscript link: https://glitch.manuscript.com/f/cases/3328282/

## GIF/Screenshots
### add project to collection
![collection-add-project](https://user-images.githubusercontent.com/938722/55996721-3967fc80-5c86-11e9-932e-fab173b6d01c.png)
### full-size collection
![collection-full](https://user-images.githubusercontent.com/938722/55996724-3967fc80-5c86-11e9-8fcb-01e1d66aaff9.png)
### collection in search results
(turn on algolia search to enable collections in search results)
![collection-search-item](https://user-images.githubusercontent.com/938722/55996725-3967fc80-5c86-11e9-91a5-2cfca4ce34ed.png)
### featured collection
![collection-wide](https://user-images.githubusercontent.com/938722/55996726-3967fc80-5c86-11e9-97e9-072ba7100723.png)
### project profile
![project-profile](https://user-images.githubusercontent.com/938722/55996727-3967fc80-5c86-11e9-9fbd-81e57fe7751a.png)
### project in search results
(turn on algolia search to enable project teams in search results)
![project-search-item](https://user-images.githubusercontent.com/938722/55996728-3967fc80-5c86-11e9-8ef3-db0a37daf6c3.png)
### project in search results, with options
![project-search-item-with-options](https://user-images.githubusercontent.com/938722/55996729-3967fc80-5c86-11e9-89a4-ff30c336bb7a.png)
### team profile
![team-profile](https://user-images.githubusercontent.com/938722/55996730-3967fc80-5c86-11e9-91d8-1b1889538168.png)
### team in search results
![team-search-item](https://user-images.githubusercontent.com/938722/55996731-3967fc80-5c86-11e9-9160-25a105e13257.png)
### user profile
![user-profile](https://user-images.githubusercontent.com/938722/55996732-3a009300-5c86-11e9-8a2d-4c3f1e7c6d74.png)

## Changes

### ProfileList component
* renders both users and teams when applicable
* supports both the condensed style used in UserList and the spaced style used in TeamList
* supports both linked and non-linked avatars
* displays a placeholder when users/teams are loading
* only displays avatars that are fully visible, instead of clipping at the edges

### Other changes
* replace & delete all instances of UserList & TeamList
* replace UserTile & TeamTile with ProfileItem (basically a one-element ProfileList)
* delete other unused files that were using old UserList & TeamList
* inline pending users rendering into AddTeamUsers
* fix missing project options on projects in search results

## How to test
Visit the pages enumerated in the screenshots sections, and verify that they look & behave correctly

## Feedback I'm looking for
- The name -- is there an existing or better term of art for user-or-team? "Avatar"? "Owner"?
- Is the placeholder behavior desirable everywhere, or should there be no placeholders in certain views?
- Are there places where we aren't linking to the profiles where we should (or vice versa?)
